DDCore/src/segmentations/HexGrid.cpp

```diff
@@ -0,0 +1,148 @@
+	registerParameter("offset_x", "Cell offset in X", _offsetX, 0., SegmentationParameter::LengthUnit, true);
+	registerParameter("offset_y", "Cell offset in Y", _offsetY, 0., SegmentationParameter::LengthUnit, true);
+	registerIdentifier("identifier_x", "Cell ID identifier for X", _xId, "x");
+	registerIdentifier("identifier_y", "Cell ID identifier for Y", _yId, "y");
+	registerIdentifier("stagger_keyword", "Volume ID identifier used for determining which volumes to stagger", _staggerKeyword, "layer");
+    }
+
+    /// Default constructor used by derived classes passing an existing decoder
+    HexGrid::HexGrid(const BitFieldCoder* decode) : Segmentation(decode) {
+        // define type and description
+	_type = "HexGridXY";
+	_description = "Hexagonal segmentation in the local XY-plane";
+
+	// register all necessary parameters                                                                        
+        registerParameter("stagger", "stagger mode", _stagger, 1);
+	registerParameter("side_length", "Cell size", _sideLength, 1., SegmentationParameter::LengthUnit);
+	registerParameter("offset_x", "Cell offset in X", _offsetX, 0., SegmentationParameter::LengthUnit, true);
+	registerParameter("offset_y", "Cell offset in Y", _offsetY, 0., SegmentationParameter::LengthUnit, true);
+	registerIdentifier("identifier_x", "Cell ID identifier for X", _xId, "x");
+	registerIdentifier("identifier_y", "Cell ID identifier for Y", _yId, "y");
+	registerIdentifier("stagger_keyword", "Volume ID identifier used for determining which volumes to stagger", _staggerKeyword, "layer");
+    
+    }
+
+    /// Destructor
+    HexGrid::~HexGrid() {
+    }
+
+    /// determine the position based on the cell ID
+    Vector3D HexGrid::position(const CellID& cID) const {
+        int layer= _decoder->get(cID,_staggerKeyword);
+	Vector3D cellPosition;
+	cellPosition.X = _decoder->get(cID,_xId )*1.5*_sideLength+_offsetX+_sideLength/2.;
+	cellPosition.Y = _decoder->get(cID,_yId )*std::sqrt(3)/2.*_sideLength+ _offsetY+_sideLength*std::sqrt(3)/2.;
+	if (_stagger==0)
+	  cellPosition.X+=_sideLength;
+	else if (_stagger==1)
+	  cellPosition.X+=(layer%3)*_sideLength;
+	else if (_stagger==2){
+	  switch (layer%4){
+	  case 0:
+	    cellPosition.X-=0.75*_sideLength;
+	    break;
+	  case 1:
+	    cellPosition.Y+=std::sqrt(3)/4*_sideLength;
+	    break;
+	  case 2:
+	    cellPosition.Y-=std::sqrt(3)/4*_sideLength;
+	    break;
+	  case 3:
+	    cellPosition.X+=0.75*_sideLength;
+	    break;
+	  }
+	}
+	return cellPosition;
+    }
+
+    inline double positive_modulo(double i, double n) {
+      return std::fmod(std::fmod(i,n) + n,n);
+    }
+
+    
+    /// determine the cell ID based on the position
+    CellID HexGrid::cellID(const Vector3D& localPosition, const Vector3D& /* globalPosition */, const VolumeID& vID) const {
+        CellID cID = vID ;
+	int layer= _decoder->get(cID,_staggerKeyword);
+	double _gridSizeY=std::sqrt(3)*_sideLength/2.;
```

Comment:
These variables are currently not used, but I think you can use them below?

Reply:
I just removed the lines that define the unused variables _gridSizeY and _gridSizeX.